crates/cli/src/lib.rs

```diff
@@ -139,6 +140,12 @@ pub fn build_app() -> Command {
         .value_parser(value_parser!(u32))
         .required(false);
 
+    let flag_ignore_warnings = Arg::new(FLAG_IGNORE_WARNINGS)
+        .long(FLAG_IGNORE_WARNINGS)
+        .help("Return a successful exit code even if there are warnings.")
```

Comment:
Hm, I would definitely expect a flag named `ignore-warnings` to ignore them in the sense of not printing them at all. 😅 

Could we name it something else? Like `--no-fail-on-warning` but more concise, maybe? 😄

Reply:
I agree, the name is definitely confusing. What about `--allow-warnings`?

Reply:
Or perhaps `--no-warnings-fail`

Reply:
🤔 do we actually just want it to ignore warnings and not print them? Is that useful? Desirable?

Alternatively, what if we had it emit a different nonzero exit code depending on whether it encountered only warnings vs some number of errors? That way when `zig build` calls it, it can distingiush between warnings only and treat that as success.

Reply:
Different exit codes sound good to me! That way we don't need to clutter the cli with another flag but the capability is still there. I will go with 
1 -> errors
2 -> warnings
unless you see reason to use something else.

Reply:
Sounds good, I'm not sure how to do that with zig build rn, but worth investigating I think.

Reply:
@rtfeldman Updated the PR; I'm not sure where a good place to document this is (if there is one). Maybe it's a niche enough use case that it doesn't need to be documented.

Reply:
Maybe we can document it in a future guide about developing platforms, since that's the only known use case we have for it!

Reply:
This isn't actually useful to the current zig build system (to my understanding).

The zig build system can expect a specific exit code, but it expects that exact exit code. So you are stuck with either always expecting 2 or always expecting 0. You can't have both.

Reply:
Can we take the exit code and wrap it in something that converts 2 to 0? Either on the zig side or (less ideally) a wrapper process?

In general I prefer this exit code approach to introducing another CLI flag if we can find a way to make it work! 😄